src/core/nixl_agent.cpp

```diff
@@ -368,29 +442,35 @@ nixlAgent::makeXferReq (const nixl_xfer_op_t &operation,
                         nixlXferReqH* &req_hndl,
                         const nixl_opt_args_t* extra_params) const {
 
-    req_hndl     = nullptr;
-    int desc_count = (int) local_indices.size();
-    nixl_opt_b_args_t opt_args;
+    nixl_opt_b_args_t  opt_args;
+    nixl_status_t      ret;
+    int                desc_count = (int) local_indices.size();
+    nixlBackendEngine* backend    = nullptr;
+
+    req_hndl = nullptr;
 
     if (!local_side || !remote_side)
         return NIXL_ERR_INVALID_PARAM;
 
     if ((!local_side->isLocal) || (remote_side->isLocal))
         return NIXL_ERR_INVALID_PARAM;
 
-    // TODO: add support for more than single backend option
-    if ((local_side->descs.size() != 1) || (remote_side->descs.size() != 1))
-        return NIXL_ERR_NOT_SUPPORTED;
-
-    // TODO: support more than single backend coming from prepXferDlist
-    nixlBackendEngine* local_backend  = local_side->descs.begin()->first;
-    nixlBackendEngine* remote_backend = remote_side->descs.begin()->first;
+    for (auto & loc_bknd : local_side->descs) {
```

Comment:
I am not sure I understand this.
So we are looking through the descriptor lists and deciding which is the backends they have registered and chose if they are both registered with the same backend? What if there are multiple backends for same combination?
Like for VRAM to VRAM (can be done by both UCX and UCX_MO)?

Reply:
So if they don't specify the first one is what we select correct?